main/ZgatewayPilight.ino

```diff
@@ -40,13 +40,18 @@ void pilightCallback(const String& protocol, const String& message, int status,
                      size_t repeats, const String& deviceID) {
   if (status == VALID) {
     Log.trace(F("Creating RF PiLight buffer" CR));
-    StaticJsonBuffer<JSON_MSG_BUFFER> jsonBuffer;
-    JsonObject& RFPiLightdata = jsonBuffer.createObject();
-    StaticJsonBuffer<JSON_MSG_BUFFER> jsonBuffer2;
-    JsonObject& msg = jsonBuffer2.parseObject(message);
-    RFPiLightdata.set("message", msg);
-    RFPiLightdata.set("protocol", (char*)protocol.c_str());
-    RFPiLightdata.set("length", (char*)deviceID.c_str());
+    StaticJsonDocument<JSON_MSG_BUFFER> jsonBuffer;
+    JsonObject RFPiLightdata = jsonBuffer.to<JsonObject>();
+    StaticJsonDocument<JSON_MSG_BUFFER> jsonBuffer2;
+    JsonObject msg = jsonBuffer2.to<JsonObject>();
```

Comment:
Should this be after the deserialization instead of before?

Reply:
In version 6 the call to `jsonBuffer2.to<JsonObject>();` clears the document. So we have to convert it before parsing otherwise it would be empty.